### ISSUE TITLE
Oauth login to fabric8 console on openstack origin local install 3.1 

### ISSUE THREAD

**Issue by iguyking** [fabric8io/fabric8#5659]
I have setup Openstack on a local install across several systems with a working high availability router.

Setup:
* Openstack Origin 3.1 (latest as of 15 Jan 2016)
* Master node is at masterorigin.mydomain.com at 192.168.1.30
* Wildcard DNS setup for default apps built in openshift routes is configured to *.apps.mydomain.com & apps.mydomain.com are pointed to 192.168.1.40

After running all the setup commands using gofabric8 and I go to the webpage of fabric8.apps.mydomain.com and the header/top UI shows up saying fabric8, then it attempts to go to the oauthclient connection and I get a connection timed out/ERR_CONNECTION_TIMED_OUT trying to get to apps.mydomain.com.  

I dug into the oauthclient setup and updated the replicationcontroller for fabric8 settings for spec/containers/env/name of OAUTH_AUTHORIZE_URI to point to masterorigin.mydomain.com and delete the existing pod to redeploy.   

This time when I go to the web page fabric8.apps.mydomain.com it redirects to login.  I login (looks a lot like the default openshift origin page) and it goes back to the fabric8 page which immediately turns around to login again.  

What am I doing wrong in the setup here?

**Comment by rawlingsj** [comment]
Hi - I'm guessing you mean Openshift rather than OpenStack (I make that mistake a lot :)).

So I think this is related to another issue to do with how gofabric8 adds the oauthclients.  I think @christian-posta has spotted the same problem.  I'll see if I can find the other issue.

**Comment by rawlingsj** [comment]
https://github.com/fabric8io/fabric8/issues/5630

**Comment by iguyking** [comment]
Yes.  Openshift, sorry.   Is there a way I can help test that the oauthclient for fabric8 is working?   How can we break this down to figure the cause out more and a workaround for the time being?

**Comment by rawlingsj** [comment]
You can try to check the oauthclients..

```
oc get oauthclient fabric8 -o=yaml > oauthclient.yml

```
make sure there's http(s) redirect entries which include your, e.g.

```
apiVersion: v1
kind: OAuthClient
metadata:
  creationTimestamp: null
  name: fabric8
redirectURIs:
- http://localhost:9090
- http://localhost:2772
- http://localhost:9000
- http://fabric8.apps.mydomain.com
- https://fabric8.apps.mydomain.com
```

then apply it back
```
oc update -f oauthclient.yml
```

Then try to access the console again.

**Comment by iguyking** [comment]
Yup.   That looks correct per everything I've seen so far.  

```yaml
apiVersion: v1
kind: OAuthClient
metadata:
  creationTimestamp: 2016-01-26T17:10:55Z
  name: fabric8
  resourceVersion: "159938"
  selfLink: /oapi/v1/oauthclients/fabric8
  uid: c2af5de9-c44f-11e5-b8a8-005056965972
redirectURIs:
- http://localhost:9090
- http://localhost:2772
- http://localhost:9000
- http://fabric8.apps.mydomain.com
- https://fabric8.apps.mydomain.com
```

Didn't make a difference.   Other places to look for why OAuth might not be working?

**Comment by christian-posta** [comment]
I've definitely seen this before. Can't recall what I did to fix this? maybe add CORS settings in the master-config.yaml file? 

what version of fabric8 console are you installing? just the latest with the gofabric8 tool?

**Comment by dsimansk** [comment]
@christian-posta yes, you're correct `corsAllowedOrigins` is needed. I've managed to access console after adding dns record of fabric8 console. Thanks!

**Comment by iguyking** [comment]
Alright.. Thanks.  That was missing in my steps and setup.   I added corsAllowedOrigins to it.  

I am using the latest gofabric8 version.  

Now when I log in I see the header line, just nothing to click on and drop down on the user in the upper right doesn't work.

**Comment by dsimansk** [comment]
@iguyking I've noticed slower loading of elements in Firefox compared to Chrome. Maybe try to switch browser to Chrome.

**Comment by iguyking** [comment]
@dsimansk I'm on Chrome.   The Chrome debug console says I can't get to the /api/v1/namespaces call.    Not sure if that's a red herring.

**Comment by lypht** [comment]
I was able to resolve by declaring two redirect uri's to the whitelist, so I have both https://fabric8.mydomain.com and https://master.mydomain.com (I deleted other entries... YMMV).  More details in #5630.  Hope that helps.

**Comment by jimmidyson** [comment]
You shouldn't need the `master.mydomain.com` entry & be careful that you're redirecting with proper scheme `https` vs `http`.

**Comment by lypht** [comment]
Ah.  Thanks for clarifying.  I removed the master, and am still resolving.
